bundles/core/org.eclipse.smarthome.core.transform/src/main/java/org/eclipse/smarthome/core/transform/internal/profiles/TransformationProfile.java

```diff
@@ -0,0 +1,137 @@
+    @NonNullByDefault({})
+    private final String format;
+
+    private final ProfileCallback callback;
+
+    public TransformationProfile(ProfileCallback callback, ProfileContext context, TransformationService service,
+            ProfileTypeUID typeUID) {
+
+        this.service = service;
+        this.profileTypeUID = typeUID;
+        this.callback = callback;
+
+        Object paramValue = context.getConfiguration().get(PATTERN_PARAM);
+        logger.debug("Value for parameter '{}' is '{}'", PATTERN_PARAM, paramValue);
+        if (paramValue instanceof String) {
+            String pattern = (String) paramValue;
+            if (pattern.contains(TransformationHelper.FUNCTION_VALUE_DELIMITER)) {
+                function = pattern.split(TransformationHelper.FUNCTION_VALUE_DELIMITER)[0];
+                format = pattern.split(TransformationHelper.FUNCTION_VALUE_DELIMITER)[1];
+            } else {
+                logError();
+                function = null;
+                format = null;
+            }
+        } else {
+            logError();
+            function = null;
+            format = null;
+
+        }
+    }
+
+    private void logError() {
+        logger.error("Parameter '{}' has to be a String containing separator '{}'. Profile will be inactive.",
+                PATTERN_PARAM, TransformationHelper.FUNCTION_VALUE_DELIMITER);
+    }
+
+    @Override
+    public ProfileTypeUID getProfileTypeUID() {
+        return profileTypeUID;
+    }
+
+    @Override
+    public void onStateUpdateFromItem(State state) {
+        callback.handleCommand((Command) state);
+    }
+
+    @Override
+    public void onCommandFromItem(Command command) {
+        callback.handleCommand(command);
+    }
+
+    @Override
+    public void onCommandFromHandler(Command command) {
+        if (function == null || format == null) {
+            return;
+        }
+        callback.sendCommand((Command) transformState(command));
+    }
+
+    @Override
+    public void onStateUpdateFromHandler(State state) {
+        if (function == null || format == null) {
+            return;
+        }
+        callback.sendUpdate((State) transformState(state));
+    }
+
+    private Type transformState(Type state) {
+        if (state instanceof StringType) {
+            String result = state.toFullString();
+            try {
+                String newFormat = String.format(format, state);
+                result = service.transform(function, newFormat);
+            } catch (TransformationException e) {
+                logger.debug("Could not transform state '{}' with function '{}' and format '{}'", state, function,
+                        format);
+            }
+            StringType resultType = new StringType(result);
+            logger.debug("Transformed '{}' into '{}'", state, resultType);
+            return resultType;
+        } else {
+            logger.debug("State '{}' is not of type StringType, skipping transformation.", state);
```

Comment:
The limitation to StringTypes makes it quite use-less in reality, doesn't it? AFAIK, transformations can be applied to all sorts of item types, not only StringItems.